File: assignments/a1/q9/Python/RuleTok.py
#RuleTok:top#
#RuleTok:import#

# <rule>RuleTok ::= <token_key> <IDENTIFIER>
class RuleTok(Rule): #RuleTok:class#

    className = "RuleTok"
    ruleString = "<rule>RuleTok ::= <token_key> <IDENTIFIER>"
    token_key = None
    identifier = None

    def __init__(token_key, identifier):
        #RuleTok:init#
        self.token_key = token_key
        self.identifier = identifier

#RuleTok#

File: assignments/a1/q9/Java/RuleTok.java
//RuleTok:top//
//RuleTok:import//
import java.util.*;

// <rule>RuleTok ::= <token_key> <IDENTIFIER>
public class RuleTok extends Rule /*RuleTok:class*/ {

    public static final String $className = "RuleTok";
    public static final String $ruleString =
        "<rule>RuleTok ::= <token_key> <IDENTIFIER>";

    public Token_key token_key;
    public Token identifier;

    public RuleTok(Token_key token_key, Token identifier) {
//RuleTok:init//
        this.token_key = token_key;
        this.identifier = identifier;
    }

    public static RuleTok parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<rule>RuleTok", scn$.lno);
        Token_key token_key = Token_key.parse(scn$, trace$);
        Token identifier = scn$.match(Token.Match.IDENTIFIER, trace$);
        return new RuleTok(token_key, identifier);
    }

    public String toString() {
        return "token " + token_key.toString() + " " + identifier;
    }

//RuleTok//
}
